File: Recursion and Backtracking/RecursionInArray/displayArrayRev.cpp
#include <iostream>
using namespace std;

void displayArrayRev(int arr[], int idx, int n){
    
    if(idx==n) return;
    
    displayArrayRev(arr,idx+1,n);
    
    cout<<arr[idx]<<endl;
}

int main(){
    int n;
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; i++)
        cin >> arr[i];
    displayArrayRev(arr, 0, n);
}

File: Recursion and Backtracking/RecursionInArray/displayArrayRev.java
import java.util.Scanner;

public class displayArrayRev {

    public static void main(String[] args) throws Exception {
        Scanner scn = new Scanner(System.in);
        int n = scn.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
            arr[i] = scn.nextInt();
        displayArrReverse(arr, 0, n);
    }

    public static void displayArrReverse(int[] arr,int idx, int n) {
        if (idx == n) return;

        displayArrReverse(arr, idx + 1, n);
        
        System.out.println(arr[idx]);
    }

}